report-generator/src/main/java/org/jacoco/core/internal/analysis/CachingInstructionsBuilder.java

```diff
@@ -0,0 +1,236 @@
+
+	/**
+	 * List of all jumps within the control flow. We need to store jumps
+	 * temporarily as the target {@link Instruction} may not been known yet.
+	 */
+	private final List<Jump> jumps;
+
+
+	/**
+	 * Creates a new builder instance which can be used to analyze a single
+	 * method.
+	 * <p>
+	 * ADDED ClassCoverageLookup classCoverageLookup parameter
+	 * REMOVED final boolean[] probes
+	 *
+	 * @param classCoverageLookup cache of the class' probes
+	 */
+	public CachingInstructionsBuilder(ClassCoverageLookup classCoverageLookup) {
+		super(null);
+		this.classCoverageLookup = classCoverageLookup;
+		this.currentLine = ISourceNode.UNKNOWN_LINE;
+		this.currentInsn = null;
+		this.instructions = new HashMap<>();
+		this.currentLabel = new ArrayList<>(2);
+		this.jumps = new ArrayList<>();
+	}
+
+	/**
+	 * Sets the current source line. All subsequently added instructions will be
+	 * assigned to this line. If no line is set (e.g. for classes compiled
+	 * without debug information) {@link ISourceNode#UNKNOWN_LINE} is assigned
+	 * to the instructions.
+	 */
+	void setCurrentLine(final int line) {
+		currentLine = line;
+	}
+
+	/**
+	 * Adds a label which applies to the subsequently added instruction. Due to
+	 * ASM internals multiple {@link Label}s can be added to an instruction.
+	 */
+	void addLabel(final Label label) {
+		currentLabel.add(label);
+		if (!LabelInfo.isSuccessor(label)) {
+			noSuccessor();
+		}
+	}
+
+	/**
+	 * Adds a new instruction. Instructions are by default linked with the
+	 * previous instruction unless specified otherwise.
+	 */
+	void addInstruction(final AbstractInsnNode node) {
+		final Instruction insn = new Instruction(currentLine);
+		final int labelCount = currentLabel.size();
+		if (labelCount > 0) {
+			for (int i = labelCount; --i >= 0; ) {
+				LabelInfo.setInstruction(currentLabel.get(i), insn);
+			}
+			currentLabel.clear();
+		}
+		if (currentInsn != null) {
+			currentInsn.addBranch(insn, 0);
+		}
+		currentInsn = insn;
+		instructions.put(node, insn);
+	}
+
+	/**
+	 * Declares that the next instruction will not be a successor of the current
+	 * instruction. This is the case with an unconditional jump or technically
+	 * when a probe was inserted before.
+	 */
+	void noSuccessor() {
+		currentInsn = null;
+	}
+
+	/**
+	 * Adds a jump from the last added instruction.
+	 *
+	 * @param target jump target
+	 * @param branch unique branch number
+	 */
+	void addJump(final Label target, final int branch) {
+		jumps.add(new Jump(currentInsn, target, branch));
+	}
+
+	/**
+	 * Adds a new probe for the last instruction.
+	 *
+	 * @param probeId index in the probe array
+	 * @param branch  unique branch number for the last instruction
+	 */
+	void addProbe(final int probeId, final int branch) {
+//		REMOVED check of probes array and instead add the probes unconditionally
+//		final boolean executed = probes != null && probes[probeId];
+		currentInsn.addBranch(true, branch);
```

Comment:
is this line also added by us? if so please explain what this does and why it's needed. if not, please change the order of this and the next instruction and add a blank line to make it clear that this is unrelated to the comment